app/src/main/java/br/org/cesar/knot_setup_app/fragment/thingRegistered/ThingFragment.java

```diff
@@ -0,0 +1,64 @@
+package br.org.cesar.knot_setup_app.fragment.thingRegistered;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ListView;
+
+import java.util.List;
+
+import br.org.cesar.knot_setup_app.R;
+
+import br.org.cesar.knot_setup_app.domain.adapter.ThingAdapter;
+import br.org.cesar.knot_setup_app.fragment.thingRegistered.ThingContract.Presenter;
+import br.org.cesar.knot_setup_app.fragment.thingRegistered.ThingContract.ViewModel;
+import br.org.cesar.knot_setup_app.model.Gateway;
+
+public class ThingFragment extends Fragment implements ViewModel {
+
+    private ListView deviceListView;
+
+    private ThingAdapter adapter;
+    private Presenter presenter;
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_list, container, false);
+
+        presenter = new ThingPresenter(this, getActivity());
+        deviceListView = view.findViewById(R.id.list);
+
+        return view;
+    }
+
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        presenter.onResume();
+    }
+
+
+
+    /**
```

Comment:
Do the javadoc in the contract. 
Doing so, if you re-define the method anywhere else, you won't need to rewrite the javadoc

Reply:
Sure, makes sense, but maybe(?) for a future PR.

Reply:
ACK.